### ISSUE TITLE
Adjust unnecessarily precise description of superfluous-parens message

### ISSUE THREAD

**Issue by mcaryl-verimatrix** [PyCQA/pylint#6226]
### Bug description

My initial code was:
```
if (i - 0) in exclude:
    continue
````
Which trigger superfluous-parens. I think the parentheses are technically unnecessary but it doesn't seem match with the intended target of this warning:
```
if (x == y):
    pass
```
I adjusted my code to:
```
foo = True
if foo and (i - 0) in exclude:
    continue
``` 
Which does not trigger superfluous-parens.
This makes me think that the warning is reacting to the proximity of the `if` and the `(` not the context in which they are being used.

### Command used

```shell
python3 -m pylint --exit-zero --rcfile=pylint-config.txt scripts/tamper.py
```

### Pylint output

```shell
scripts/tamper.py:152:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
```

### Expected behavior

No superfluous-parens warning.

### Pylint version

```shell
pylint 2.12.2
```

### OS / Environment

macOS Monterey 12.2.1

**Comment by Pierre-Sassoulas** [comment]
I think it's fixed in 2.13.5, could you confirm ?

**Comment by mcaryl-verimatrix** [comment]
I was using an old version... more testing...

It still seems to be a problem `C0325: Unnecessary parens after 'if' keyword (superfluous-parens)` with `pylint 2.13.5`.

**Comment by Pierre-Sassoulas** [comment]
Could you show the content of pylint-config.txt please ? I cannot reproduce.

**Comment by jacobtylerwalls** [comment]
I can reproduce it. But why is it a false positive?

**Comment by mcaryl-verimatrix** [comment]
I wasn't sure of the exact rules. My assumption was that if `if foo and (i - 0) in exclude` doesn't trigger a warning then `if (i - 0) in exclude` should also not trigger a warning.

**Comment by mcaryl-verimatrix** [comment]
From the pylint code `_check_keyword_parentheses` there is a comment:
```
Parens are unnecessary if there is exactly one balanced outer pair on a
line, and it is followed by a colon, and contains no commas (i.e. is not a
tuple).
```
This does not seem to match `if (i - 0) in exclude`.

Hmm. Looking further down `_check_keyword_parentheses` I can see a line starting `if tokens[i + 1].string in {":", ")", "]", "}", "in"}...`. The inclusion of `in` in this list is presumably why I'm seeing a warning for one line. And the variable `found_and_or` is presumably why I'm not seeing a warning for the other line.

 I haven't followed through all the logic `_check_keyword_parentheses` and I don't know the precise definition of "superfluous". So this is probably doing what is intended, just not what I expected. Maybe the documentation, https://vald-phoenix.github.io/pylint-errors/plerr/errors/format/C0325.html, should be updated?

In any case, many thanks for the prompt response.

**Comment by Pierre-Sassoulas** [comment]
>  Maybe the documentation, https://vald-phoenix.github.io/pylint-errors/plerr/errors/format/C0325.html, should be updated?

This repository is maintained by @vald-phoenix who is currently busy fighting in Ukraine. But we're integrating it in our own documentation with his permission currently, see #5953. If you want to provide an example for https://pylint.pycqa.org/en/latest/messages/convention/superfluous-parens.html it would be very much appreciated  :)

**Comment by jacobtylerwalls** [comment]
I think you have a good point that the message description is too specific about what it is looking for. The `followed by a colon` should be struck.

**Comment by mpernigo** [comment]
Hey! If that's ok with you, I could work on these changes.
I'll put up 2 PRs for:
1. Updating the docstring
2. Adding examples to the doc

**Comment by Pierre-Sassoulas** [comment]
Thank you ! I assigned you to the issue :)
